### ISSUE TITLE
Table.read() error when reading ECSV using astropy dev

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#6003]
_Issue [JP-2068](https://jira.stsci.edu/browse/JP-2068) was created on JIRA by [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies):_

We are seeing a failure in our weekly cron job of unit tests using dev dependencies.  This one is because of some changes to the ECSV format in astropy dev:

<https://github.com/spacetelescope/jwst/runs/2489783314?check_suite_focus=true#step:9:51414>

The failing tests in the `jwst` repo are:

```
test_calc_deltas
test_from_models
```

failing with the same error in `Table.read()` when reading an ECSV file:

```
E               ValueError: datatype 'object' of column 'exposure' is not in allowed values ('bool', 'int8', 'int16', 'int32', 'int64', 'uint8', 'uint16', 'uint32', 'uint64', 'float16', 'float32', 'float64', 'float128', 'string')
```

Also, the test should be using a context manager when opening the ECSV file using `Table.read()`, hence all the open-files ERRORs, though clearly these 2 tests are contaminating others, somehow.

**Comment by stscijgbot-jp** [comment]
_Comment by [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies) on [JIRA](https://jira.stsci.edu/browse/JP-2068?focusedCommentId=530393#comment-530393):_

Fixed in #6090

**Comment by jdavies-st** [comment]
Issue filed https://github.com/astropy/astropy/issues/11803
